src/Utils.ts

```diff
@@ -249,19 +244,66 @@ export namespace Utils {
         return mavenPath ? `"${mavenPath}"` : "mvn";
     }
 
-    export function executeMavenCommand(command: string, pomfile: string): void {
+    function getMaven(): string {
+        const executablePathInConf: string = workspace.getConfiguration("maven.executable").get<string>("path");
+        // TODO: use maven wrapper if path is not set and mvnw is found.
+        return executablePathInConf ? executablePathInConf : "mvn";
+    }
+
+    function wrapMaven(mvn: string): string {
+        if (mvn === "mvn") {
+            return mvn;
+        } else {
+            return `"${mvn}"`;
+        }
+    }
+
+    export function executeInTerminal(command: string, pomfile: string): void {
+        const mvnString: string = wrapMaven(getMaven());
         const fullCommand: string = [
-            Utils.getMavenExecutable(),
+            mvnString,
             command.trim(),
             "-f",
             `"${formattedFilepath(pomfile)}"`,
             workspace.getConfiguration("maven.executable", Uri.file(pomfile)).get<string>("options")
         ].filter((x: string) => x).join(" ");
-        const name: string = "Maven";
+        const workspaceFolder: WorkspaceFolder = workspace.getWorkspaceFolder(Uri.file(pomfile));
+        const name: string = workspaceFolder ? `Maven-${workspaceFolder.name}` : "Maven";
         VSCodeUI.runInTerminal(fullCommand, { name });
         updateLRUCommands(command, pomfile);
     }
 
+    export async function executeInBackground(command: string, pomfile: string): Promise<{}> {
+        const mvnString: string = wrapMaven(getMaven());
+
+        const fullCommand: string = [
+            mvnString,
+            command.trim(),
+            "-f",
+            `"${formattedFilepath(pomfile)}"`,
+            workspace.getConfiguration("maven.executable", Uri.file(pomfile)).get<string>("options")
+        ].filter((x: string) => x).join(" ");
+
+        const rootfolder: WorkspaceFolder = workspace.getWorkspaceFolder(Uri.file(pomfile));
+        const customEnv: {} = VSCodeUI.setupEnvironment();
+        const execOptions: child_process.ExecOptions = {
+            cwd: rootfolder ? rootfolder.uri.fsPath : path.dirname(pomfile),
```

Comment:
Why we need path.dirname(pomfile) here?

Reply:
it was to fix mvnw issue months ago. Actually it should be `path.dirname(path of mvnw)`. Will force to use maven wrapper if found.